### PR TITLE
updating releases page to reflect release windows

### PR DESCRIPTION
**Pull request by CEHENKLE**
### Description
updated release page to reflect https://github.com/opensearch-project/.github/issues/150
 
### Issues Resolved
https://github.com/opensearch-project/.github/issues/150

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by CEHENKLE** [comment]
@bbarani useful?

File: releases.md [hunk 1]
```diff
@@ -16,39 +16,30 @@ _Updated April 26, 2023_
...
+For minor version releases, OpenSearch follows "release window" model as described in this [proposal](https://github.com/opensearch-project/.github/issues/150).  The goal is to release a new minor version approximately every six weeks which includes all the new features and fixes that are ready to go. Once we enter a release window, every day we will generate a release candidate. When the exit criteria are met by a release candidate, we will make an announcement, update the release date column above and publish new artifacts on that date. 
 
-**Note:** If we cannot pass the exit criteria by 2 weeks after the start of the release cycle window, we will cancel the minor release and hold changes until the next release window.
+**Note:** If we cannot pass the exit criteria by 2 weeks after the start of the release cycle window (1 week for 1.x releases), we will cancel the minor release and hold changes until the next release window.
```
Comment (bbarani) [inline releases.md:38]: We only do patch release in 1.x line. I assume the updated release process is only valid for minor and patch release will still follow release train model?

File: releases.md [hunk 2]
```diff
@@ -16,39 +16,30 @@ _Updated April 26, 2023_
...
-| 1.3.8         | January 26th     | February 2nd   |
-| 2.6.0         | February 21th    | February 28th  |
-| 1.3.9         | March 9th        | March 16th     |
-| 2.7.0         | April 17th       | May 2nd        |
-| 1.3.10        | May 11th         | May 18th       |
-| 2.8.0         | May 30th         | June 6th       |
-| 1.3.11        | June 22nd        | June 29th      |
-| 2.9.0         | July 11th        | July 24th      |
-| 1.3.12        | August 3rd       | August 10th    |
-| 2.10.0        | September 05th   | TBD            |
```
Comment (krisfreedain) [inline releases.md:37]: I don't think we want to delete this history as we move it to a 'Release History' section at the end of the year.

**Review by krisfreedain — CHANGES_REQUESTED** [review]
Left a comment on too many deletions. There's a comment from @bbarani in there as well

File: releases.md (hunk 2 above)
Comment (CEHENKLE) [reply releases.md:37]: I thought about it, and decided that most folks didn't care, and if they did, they could pull the history off the page.   We don't, for example, show all the earlier release dates like 1.0-1.3.   If you think folks would like a release history page, do you want to make one?

File: releases.md (hunk 1 above)
Comment (CEHENKLE) [reply releases.md:38]: If I updated it to being "1 week for 1.x patch releases", would that be more or less clear?

Comment (bbarani) [reply releases.md:38]: Yeah, Patch release still follows release train model so just want to make sure that no one gets confused with updated minor release process.

Comment (CEHENKLE) [reply releases.md]: okay, updated.  Please review, LMK if there are any other changes.  Thanks!

**Review by bbarani — DISMISSED** [review]

**Comment by krisfreedain** [comment]
hold - resolving issues with the table layout

**Comment by krisfreedain** [comment]
Minor edits made - LGTM

**Review by krisfreedain — APPROVED** [review]
LGTM

**Pull request merged**